ql/src/java/org/apache/hadoop/hive/ql/exec/Utilities.java

```diff
@@ -1514,6 +1523,17 @@ public static void mvFileToFinalPath(Path specPath, Configuration hconf,
     fs.delete(taskTmpPath, true);
   }
 
+  private static void createFileList(Set<FileStatus> filesKept, Path srcPath, Path targetPath, FileSystem fs)
+      throws IOException {
+    try (FSDataOutputStream outStream = fs.create(new Path(targetPath, BLOB_FILES_KEPT))) {
+      outStream.writeBytes(srcPath.toString() + System.lineSeparator());
```

Comment:
Is it needed to write the srcPath as first entry?  Only file paths can be written out and during reading, first entry's parent directory can be assumed as srcPath?

Reply:
Have added a comment above explaining the reason, as discussed offline